src/modules/EventuateClient.js

```diff
@@ -579,11 +592,14 @@ export default class EventuateClient {
   }
 
   makeEvent(eventStr, ack) {
-
     try {
+      const parsedEvent = JSON.parse(eventStr);
+      const { id: eventId, eventType, entityId, entityType, swimlane, eventToken } = parsedEvent;
+      let { eventData: eventDataStr } = parsedEvent;
 
-      const { id: eventId, eventType, entityId, entityType, eventData: eventDataStr, swimlane, eventToken } = JSON.parse(eventStr);
-
+      if (this.encryption && eventDataStr.includes(this.encryption.prefix)) {
```

Comment:
Move the `if (this.encryption && eventDataStr.includes(this.encryption.prefix))` into `this.decrypt()`